### DISCUSSION TITLE
[3.0] Make the package framework-agnostic

### DISCUSSION THREAD

**Discussion by stefanzweifel** [stefanzweifel/laravel-stats#153] [Ideas]
As the name of this repository might suggest, *stats* currently only works for Laravel applications.

Over the years I've received multiple questions, if *stats* could be made framework agnostics so it could be used in other frameworks too (for example Symfony applications).  
With the work done in the release of 2.0 such a step should be easier now. The package basically consists of the following parts:

- A Class Collector: Loops through folders in a path and finds all PHP classes
- Value Objects to describe a Project: A `Project`-instance takes a collection of Classes and Classifies them.
- A collection of Classifiers: These simple classes take a given class in a Laravel project and try to associate them with a "Component"
- Laravel specific classes: Artisan Command, ServiceProvider, etc.

My idea to make *stats* framework agnostic, is to move the Class Collector and the Value Objects into a separate package. This package (`laravel-stats`) would require this new package as a dependency and would pass its list of Laravel specific Classifiers to the newly created "core"-package.    
It would then be trivial to create separate packages for other frameworks like Symfony or Zend: Define your own list of Classifiers which describe components in each framework and pass them to the "core"-package.

To make this actually work, the Class Collector and Value Obejcts would have to be rewritten though. They currently rely heavily on Laravels Collections.   
The new "core"-package probably shouldn't have `illuminate/support` as a dependency and these mentioned classes should then rather use plain older PHP arrays.

**If you're interested in this idea, please share your comments, opinions and ideas below!**

**Comment by djunehor** [comment]
I think it'll be nice, so applications running other PHP frameworks can benefit. Let the core be framework-agnostic. Then we can create classifiers for various frameworks, grouped in folders, so it's easy to create classifiers for other frameworks or php applications
